source/routing/raptor_api.cpp

```diff
@@ -51,6 +51,126 @@ namespace navitia { namespace routing {
+ *                       \ Y
+ *      total_nb_journeys < min_nb_journeys?
+ *                   /        \
+ *                  /Y         \N
+ *              Case 2       Case 3
+ *
+ *
+ *
+ */
+static bool keep_going(const uint32_t total_nb_journeys,
+                       const uint32_t nb_try,
+                       const bool clockwise,
+                       const DateTime request_date_secs,
+                       const boost::optional<uint32_t>& min_nb_journeys,
+                       const boost::optional<uint32_t>& timeframe_duration) {
+
+    auto is_inside = [clockwise](DateTime lhs, DateTime rhs){
+        return clockwise ? lhs < rhs : lhs > rhs;
+    };
+    if (min_nb_journeys) {
+        if (! timeframe_duration) {
+            // Case 1: we return all min_nb_journeys journeys that raptor can find
+            return (total_nb_journeys < min_nb_journeys.get()) && (nb_try <= max_nb_raptor_call);
+        }
+        if (total_nb_journeys < min_nb_journeys.get()) {
+            // Case 2: we continue searching journeys until min_nb_journeys is reached
+            return true;
+        }
+        // Case 3: we have already enough journeys, but the search datetime is still inside of the time frame
+        //         we keep searching until the end of time frame
+        return is_inside(request_date_secs, timeframe_duration.get());
+    }
+    if (! timeframe_duration) {
+        // Case 4: Neither time duration nor min_nb_journeys are given, we call only once Raptor
+        return false;
+    }
+    // Case 5: no min_nb_journeys is given, we find all journeys inside of the time frame
+    return is_inside(request_date_secs, timeframe_duration.get());
+}
+
+/**
+ * @brief Culling excessive journeys if we exceed limits
+ */
+static void culling_excessive_journeys(const boost::optional<uint32_t>& min_nb_journeys,
+                                       const boost::optional<uint32_t>& timeframe_duration,
+                                       const bool clockwise,
+                                       JourneySet& journeys) {
+    log4cplus::Logger logger = log4cplus::Logger::getInstance(LOG4CPLUS_TEXT("logger"));
+
+    if (timeframe_duration && journeys.size() > 0) {
+
+        routing::JourneyCmp journey_cmp{clockwise};
+        // filter journeys by end datetime until min_nb_journeys
+        std::vector<routing::Journey> sorted{journeys.begin(), journeys.end()};
+        std::sort(sorted.begin(), sorted.end(), journey_cmp);
+
+        auto is_inside = [clockwise](const routing::Journey& j, DateTime dt){
+            return clockwise ? j.departure_dt < dt : j.arrival_dt > dt;
+        };
+
+        uint32_t count = 0;
+        boost::for_each(sorted, [&](const routing::Journey& j) {
+
+            // Erase journey if
+            // min_nb_journeys is deactivated and journey exceed the time frame limit
+            // or
+            // min_nb_journeys exist and Nb journeys is greater than min_nb_journeys criteria
```

Comment:
Shouldn't we do the culling when we exceed the `max_nb_journeys` (rather than the `min_nb_journey`) ?